src/safe_mpc/abort.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+from casadi import SX, dot
+from .abstract import AbstractController
+
+
+class SafeBackupController(AbstractController):
```

Comment:
From my point of view, `AbstractController` can refer both to single OCP or MPC, since the initialization of the Acados OCP object is almost equal. To solve an OCP, one can set the initial guess and call the method `solve`, while for the MPC cycle we call `step`, which is defined differently for the controllers. In this way, we can avoid duplicated code fragments.

Reply:
ok